src/pages/waiting-room/cat-home-test/__tests__/waiting-room.cat-d.page.spec.ts

```diff
@@ -0,0 +1,291 @@
+        component.ionViewDidEnter();
+        expect(screenOrientation.lock)
+          .toHaveBeenCalledWith(screenOrientation.ORIENTATIONS.PORTRAIT_PRIMARY);
+      });
+
+      it('should keep the device awake', () => {
+        component.ionViewDidEnter();
+        expect(insomnia.keepAwake).toHaveBeenCalled();
+      });
+
+    });
+
+    describe('clickBack', () => {
+      it('should should trigger the lock screen', () => {
+        component.clickBack();
+        expect(deviceAuthenticationProvider.triggerLockScreen).toHaveBeenCalled();
+      });
+    });
+
+    describe('onSubmit', () => {
+      it('should navigate to the COMMUNICATION_PAGE if the form is valid', () => {
+        const formGroup = component.formGroup;
+        formGroup.addControl('insuranceCheckbox', new FormControl('', [Validators.requiredTrue]));
+        formGroup.get('insuranceCheckbox').setValue(true);
+        component.onSubmit();
+        expect(navController.push).toHaveBeenCalled();
+      });
+      it('should dispatch the WaitingRoomValidationError action if a field is not valid', fakeAsync(() => {
+        const formGroup = component.formGroup;
+        formGroup.addControl('insuranceCheckbox', new FormControl('', [Validators.requiredTrue]));
+        formGroup.get('insuranceCheckbox').setValue(false);
+        component.onSubmit();
+        tick();
+        expect(store$.dispatch).toHaveBeenCalledWith(new WaitingRoomValidationError('insuranceCheckbox is blank'));
+      }));
+    });
+
+    describe('isJournalDataInvalid', () => {
+      const journalData: CatDUniqueTypes.JournalData = {
+        examiner: {
+          staffNumber: 'real-staff-number',
+        },
+        testCentre: {
+          centreId: 11223344,
+          centreName: 'name',
+          costCode: 'cost code',
+        },
+        testSlotAttributes: {
+          slotId: 12123331,
+          start: '2019-11-11',
+          vehicleTypeCode: 'vehicle type code',
+          welshTest: true,
+          specialNeeds: true,
+          extendedTest: false,
+        },
+        candidate: {
+          candidateName: {
+            firstName: 'fname',
+            lastName: 'lname',
+          },
+        } as CatDUniqueTypes.Candidate,
+        applicationReference: {
+          applicationId: 11223344141414,
+          bookingSequence: 112,
+          checkDigit: 11,
+        },
+      };
+
+      it('should return true if no examiner staffnumber', () => {
+        const result = component.isJournalDataInvalid({
+          ...journalData,
+          examiner: {
+            staffNumber: '',
+          },
+        });
+        expect(result).toBeTruthy;
+      });
+
+      it('should return true if no candidate name & driver number', () => {
+        const result = component.isJournalDataInvalid({
+          ...journalData,
+          candidate: {
+            candidateName: {},
+            driverNumber: '',
+          } as CatDUniqueTypes.Candidate,
```

Comment:
needs a to-do

Reply:
instead of a TODO - it now uses the right type